sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -4,18 +4,635 @@
+{
+  // We can't always assume the input wide string is null-terminated, so need to pass in the actual
+  // size.
+  size_t wideStrSize = wideString.size();
+  if (wideStrSize > INT_MAX)
+  {
+    throw Azure::Core::Http::TransportException(
+        "Input wide string is too large to fit within a 32-bit int.");
+  }
+
+  int wideStrLength = static_cast<int>(wideStrSize);
+  int sizeNeeded
+      = WideCharToMultiByte(CP_UTF8, 0, wideString.c_str(), wideStrLength, NULL, 0, NULL, NULL);
+  if (sizeNeeded == 0)
+  {
+    // Errors include:
+    // ERROR_INSUFFICIENT_BUFFER
+    // ERROR_INVALID_FLAGS
+    // ERROR_INVALID_PARAMETER
+    // ERROR_NO_UNICODE_TRANSLATION
+    DWORD error = GetLastError();
+    throw Azure::Core::Http::TransportException(
+        "Unable to get the required transcoded size for the input wide string. Error Code: "
+        + std::to_string(error) + ".");
+  }
+
+  std::string str(sizeNeeded, 0);
+  if (WideCharToMultiByte(
+          CP_UTF8, 0, wideString.c_str(), wideStrLength, &str[0], sizeNeeded, NULL, NULL)
+      == 0)
+  {
+    DWORD error = GetLastError();
+    throw Azure::Core::Http::TransportException(
+        "Unable to transcode the input wide string to a string. Error Code: "
+        + std::to_string(error) + ".");
+  }
+  return str;
+}
+
+std::string WideStringToStringASCII(
+    wchar_t const* const wideStringStart,
+    wchar_t const* const wideStringEnd)
+{
+  // Converting this way is only safe when the text is ASCII.
+#pragma warning(suppress : 4244)
+  std::string str(wideStringStart, wideStringEnd);
+  return str;
+}
+
+void ParseHttpVersion(
+    const std::string& httpVersion,
+    uint16_t* majorVersion,
+    uint16_t* minorVersion)
+{
+  auto httpVersionEnd = httpVersion.data() + httpVersion.size();
+
+  // Set response code and HTTP version (i.e. HTTP/1.1)
+  auto majorVersionStart
+      = httpVersion.data() + 5; // HTTP = 4, / = 1, moving to 5th place for version
+  auto majorVersionEnd = std::find(majorVersionStart, httpVersionEnd, '.');
+  auto majorVersionInt = std::stoi(std::string(majorVersionStart, majorVersionEnd));
+
+  auto minorVersionStart = majorVersionEnd + 1; // start of minor version
+  auto minorVersionInt = std::stoi(std::string(minorVersionStart, httpVersionEnd));
+
+  *majorVersion = (uint16_t)majorVersionInt;
+  *minorVersion = (uint16_t)minorVersionInt;
+}
+
+/**
+ * @brief Add a list of HTTP headers to the @RawResponse.
+ *
+ * @remark The \p headers must contain valid header name characters (RFC 7230).
+ * @remark Header name, value and delimiter are expected to be in \p headers.
+ *
+ * @param headers The complete list of headers to be added, in the form "name:value\0",
+ * terminated by "\0".
+ *
+ * @throw if \p headers has an invalid header name or if the delimiter is missing.
+ */
+void AddHeaders(std::string const& headers, std::unique_ptr<RawResponse>& rawResponse)
+{
+  auto begin = headers.data();
+  auto end = begin + headers.size();
```

Comment:
I am dealing with pointers here rather than iterators because of where these are going to be passed to (and to do arithmetic operations). So, I don' think this will work.